### PR TITLE
CTED-2800 | Hot Question | Posted Question Visibility

### PR DESCRIPTION
**Pull request by Mmahadinana**
- restore_hotquestion_stepslib.php adding additional condition to focus only on the active round.
	- Adding the observer to update the round starttime to course startdate
	- adding events.php file to listend to module released
	- update the version.ph

### CODE REVIEW

**Comment by Mmahadinana** [comment]
file: mod/hotquestion/db/events.php
image

file: mod/hotquestion/classes/observer.php
image

file: mod/hotquestion/backup/moodle2/restore_hotquestion_stepslib.php
image

**Review by brett-fritz — APPROVED** [review]

File: backup/moodle2/restore_hotquestion_stepslib.php [hunk 1]
```diff
@@ -115,10 +115,10 @@ protected function process_hotquestion_round($data) {
         $data->starttime = $this->apply_date_offset($data->starttime);
         $data->endtime = $this->apply_date_offset($data->endtime);
         // CTED-2793 | ensure that the starttime for round is not in the future.
-        // If is in the future assigh course starttime to round starttime
-        if ($data->starttime > strtotime("+1 day")) {
+        // If is in the future assigh course starttime to round starttime.
+        if ($data->starttime > strtotime("+1 day" && $data->endtime == 0 )) {
```
Comment (BrendonPretorius) [inline backup/moodle2/restore_hotquestion_stepslib.php]: Is this correct? You are doing the && conditional in the strtotime function

File: classes/observer.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+            $param = array($event->courseid, $event->objectid);
+
+            $questionround = $DB->get_records_sql($sql, $param);
+
+            if (!empty($questionround)) {
+                foreach ($questionround as $round) {
+                    $round->starttime = $round->startdate;
+                    // Update the starttime with the course startdate.
+                    $DB->update_record("hotquestion_rounds", $round, array('id' => $round->id));
```
Comment (BrendonPretorius) [inline classes/observer.php]: Incorrect use of the update_record function. 3rd param is if it is a bulk update https://docs.moodle.org/dev/Data_manipulation_API#Updating_records

File: classes/observer.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                }
+            }
+        } catch (\Exception $e) {
+            error_log('Failed to get and update hotquestion_rounds Observer ' . $e->getMessage());
```
Comment (BrendonPretorius) [inline classes/observer.php]: Please use our normal error logging standard https://confluence.2u.com/pages/viewpage.action?spaceKey=CTED&title=Error+Log+Standards

File: classes/observer.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+ * mod_hotquestion file observers.
+ *
+ * @package    mod_hotquestion
+ * @copyright  2023 mmohohlo-marumo <>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+class mod_hotquestion_observer {
+
+    public static function hotquestion_round(\local_getsmarter\event\module_release $event) {
```
Comment (BrendonPretorius) [inline classes/observer.php:31]: Please add relevant doc blocks

**Review by BrendonPretorius — CHANGES_REQUESTED** [review]

**Review by BrendonPretorius — APPROVED** [review]

**Pull request merged**